File: sandbox-runner/patterns/mapreduce-tree/go/starter.go
package main

import (
	"context"
	"fmt"
	"log"
	"strings"
	"time"

	"go.temporal.io/sdk/client"
)

func main() {
	c, err := client.Dial(client.Options{HostPort: "localhost:7233"})
	if err != nil {
		log.Fatalln("Unable to create client:", err)
	}
	defer c.Close()

	workflowID := fmt.Sprintf("%s-%d", WorkflowIDPrefix, time.Now().UnixMilli())
	we, err := c.ExecuteWorkflow(
		context.Background(),
		client.StartWorkflowOptions{
			ID:        workflowID,
			TaskQueue: TaskQueue,
		},
		NodeWorkflow,
		Records,
		0,
		"",
	)
	if err != nil {
		log.Fatalln("Unable to execute workflow:", err)
	}
	fmt.Printf("Started workflow: %s\n", we.GetID())
	fmt.Printf("Processing %d records via MapReduce Tree…\n", len(Records))

	var results []string
	if err := we.Get(context.Background(), &results); err != nil {
		log.Fatalln("Workflow failed:", err)
	}
	fmt.Printf("MapReduce Tree complete: %d results\n", len(results))
	fmt.Printf("Results: %s\n", strings.Join(results, ", "))
	fmt.Printf(
		"Open the Temporal UI and search for '%s' to see the full workflow tree.\n",
		workflowID,
	)
}

File: sandbox-runner/patterns/mapreduce-tree/java/src/main/java/Starter.java
import io.temporal.client.WorkflowClient;
import io.temporal.client.WorkflowOptions;
import io.temporal.serviceclient.WorkflowServiceStubs;

import java.util.List;

public class Starter {
    public static void main(String[] args) {
        WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
        WorkflowClient client = WorkflowClient.newInstance(service);

        String workflowId = Shared.WORKFLOW_ID_PREFIX + "-" + System.currentTimeMillis();
        NodeWorkflow.Node workflow = client.newWorkflowStub(
                NodeWorkflow.Node.class,
                WorkflowOptions.newBuilder()
                        .setTaskQueue(Shared.TASK_QUEUE)
                        .setWorkflowId(workflowId)
                        .build());

        System.out.println("Started workflow: " + workflowId);
        System.out.println("Processing " + Shared.RECORDS.size() + " records via MapReduce Tree…");
        List<String> results = workflow.run(Shared.RECORDS, 0, "");
        System.out.println("MapReduce Tree complete: " + results.size() + " results");
        System.out.println("Results: " + String.join(", ", results));
        System.out.println(
                "Open the Temporal UI and search for '" + workflowId
                        + "' to see the full workflow tree.");
    }
}
